### PR TITLE
Adds sharpening blocks

### PR DESCRIPTION
**Pull request by PKPenguin321**
Two can be found in the kitchen vendor.

They are single-use. When a weapon is used on them, it gains the prefix "sharpened," gets the SHARP flag (allowing it to butcher things), and has it's force/throwforce increase by ~~5~~ 4 up to a maximum of ~~15~~ ~~12~~ 30. If an item already has ~~15~~ ~~12~~ 30 or higher force (or throwforce (really only energy swords will have this)), they cannot be sharpened.

This also adds an admin variant, the super sharpening block, that makes an item do 200 damage on hit/throw and gain the prefix "super-sharpened."

~~Items with block chance cannot be sharpened. Blame zilenan~~ **Only items that are already sharp can be sharpened.**

~~Yes, you can have "the sharpened sharpened rolling pin." Double prefixes can happen, but you'll only rarely see a prefix repeated once, and you should _never_ see a prefix repeated more than twice (as by the time a third block is used, you'll have hit the force damage cap). This is intentional. By maintainer request, prefixes now go sharpened -> twice-sharpened -> thrice-sharpened~~ You can no longer sharpen the same item more than once.

Two handed items get their wielded force increased rather than standard force.

Sprite by newfren.
8fn6s9w

:cl: PKPenguin321
rscadd: The kitchen vendor now contains two sharpening blocks.
rscadd: Only items that are already sharp (such as fire axes, knives, etc) can be sharpened. Items used on sharpening blocks become sharper and deadlier. The sharpening blocks themselves can only be used once, and you can only sharpen items once.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/sharpener.dm [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/obj/item/weapon/sharpener
+	name = "sharpening block"
```
Comment (GunHog) [inline code/game/objects/items/weapons/sharpener.dm]: I think you are looking for "whetstone".

**Comment by KazeEspada** [comment]
>Can't sharpen a fire axe or a cleaver.

**Comment by GunHog** [comment]
I expect this to have a similar effect to Goofball's deep fryer, in terms of me sharpening the silliest of things.

**Comment by PKPenguin321** [comment]
>I expect this to have a similar effect to Goofball's deep fryer, in terms of me sharpening the silliest of things.

Good, players should have fun with this. It's a pretty funny feature.

File: code/game/objects/items/weapons/sharpener.dm [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		user << "<span class='notice'>You sharpen [I] with [src], making it much more deadly than before.</span>"
+		I.sharpness = IS_SHARP
+		I.force = (I.force + increment)
+		I.throwforce = (I.throwforce + increment)
+		I.name = "[prefix] [I.name]"
+		src.name = "dull [src.name]"
+		src.desc = "[src.desc] At least, it used to."
+		used = 1
+		if(I.force > max)//if the item's force exceeds 15 after 5 force is added,
+			I.force = max//set it back to 15
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/sharpener.dm]: No no no, you do 

Clamp(I.force + 5, 0, 15)

**Comment by PKPenguin321** [comment]
>>Can't sharpen a fire axe or a cleaver.

tfw nerds would cut your head off while chanting "OVERPOWERED, OVERPOWERED, OVERPOWERED"

File: code/game/objects/items/weapons/sharpener.dm [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		if(I.throwforce > max)//do this again for throwforce
+			I.throwforce = max
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/sharpener.dm]: Same as above

Clamp(I.throwforce + 5, 0, 15)

File: code/game/objects/items/weapons/sharpener.dm [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/obj/item/weapon/sharpener
+	name = "sharpening block"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "recharger0"
+	desc = "A block that makes things sharp."
+	var/used = 0
+	var/increment = 5
+	var/max = 15
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/sharpener.dm]: Useless var

File: code/game/objects/items/weapons/sharpener.dm (hunk 2 above)
Comment (PKPenguin321) [inline code/game/objects/items/weapons/sharpener.dm]: just replace these two lines with that one line?

File: code/game/objects/items/weapons/sharpener.dm [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+		src.desc = "[src.desc] At least, it used to."
+		used = 1
+		if(I.force > max)//if the item's force exceeds 15 after 5 force is added,
+			I.force = max//set it back to 15
+		if(I.throwforce > max)//do this again for throwforce
+			I.throwforce = max
+
+/obj/item/weapon/sharpener/super
+	name = "super sharpening block"
+	desc = "A block that will make your weapon sharper than Einstein on adderol."
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/sharpener.dm]: adderall

File: code/game/objects/items/weapons/sharpener.dm (hunk 2 above)
Comment (KorPhaeron) [inline code/game/objects/items/weapons/sharpener.dm]: Well no replace the whole I.force = I.force+ increment with that line, and just remove the

If(I.force) stuff

**Comment by KorPhaeron** [comment]
>and you should never see a prefix repeated more than twice (as by the time a third block is used, you'll have hit the 15 force damage cap).

Until I use it 3 times on my medkit or something

File: code/game/objects/items/weapons/sharpener.dm [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+		return
+	if(I.force >= max || I.throwforce >= max)
+		user << "<span class='notice'>[I] is much too powerful to sharpen further.</span>"
+		return
+	if(istype(I, /obj/item/weapon))
+		user << "<span class='notice'>You sharpen [I] with [src], making it much more deadly than before.</span>"
+		I.sharpness = IS_SHARP
+		I.force = (I.force + increment)
+		I.throwforce = (I.throwforce + increment)
+		I.name = "[prefix] [I.name]"
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/sharpener.dm]: I.name = "[prefix] [initial(I.name)]"

Comment (KorPhaeron) [inline code/game/objects/items/weapons/sharpener.dm]: If you want to avoid multiple prefixes

**Comment by drynwyn** [comment]
You should make em' orderable at cargo or put them in the autolathe or maybe make them spawn in maint, just so that if you want to get a whole bunch and sharpen everything for DANGERSTATION, you can.

File: code/game/objects/items/weapons/sharpener.dm (hunk 6 above)
Comment (PKPenguin321) [inline code/game/objects/items/weapons/sharpener.dm]: idk man multiple prefixes is kinda silly (the fun kind) and helps to show exactly HOW sharp whatever you're getting attacked by is, i think i want to keep it. if you as a maintainer do not like it i will remove it however.

**Comment by PKPenguin321** [comment]
>Until I use it 3 times on my medkit or something

the thing is, 3 prefixes is the max. 5x3 = 15, and at that point you're not able to get another prefix. not to mention you should only be getting 2 sharpening blocks in a round (not counting away missions/janky maps that might have more than 1 kitchen for whatever reason)

File: code/game/objects/items/weapons/sharpener.dm (hunk 5 above)
Comment (PKPenguin321) [inline code/game/objects/items/weapons/sharpener.dm]: fugg

File: code/game/objects/items/weapons/sharpener.dm (hunk 1 above)
Comment (PKPenguin321) [inline code/game/objects/items/weapons/sharpener.dm]: >implying anybody knows what that is
>implying people don't know what a sharpening block is

File: code/game/objects/items/weapons/sharpener.dm (hunk 6 above)
Comment (KorPhaeron) [inline code/game/objects/items/weapons/sharpener.dm]: I have added far sillier things than double sharpened objects so I will leave that up to other people

File: code/game/objects/items/weapons/sharpener.dm [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+		src.name = "dull [src.name]"
+		src.desc = "[src.desc] At least, it used to."
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/sharpener.dm]: `src` is implied.

Also, sharpening blocks don't go dull....

**Comment by Fox-McCloud** [comment]
Sharpening of species' claws when? =p

File: code/game/objects/items/weapons/sharpener.dm (hunk 7 above)
Comment (PKPenguin321) [inline code/game/objects/items/weapons/sharpener.dm]: I knew it, I just wasn't sure.

*These* sharpening blocks dull. They're made in space-china.

File: code/game/objects/items/weapons/sharpener.dm [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+	desc = "A block that makes things sharp."
+	var/used = 0
+	var/increment = 5
+	var/max = 15
+	var/prefix = "sharpened"
+
+
+/obj/item/weapon/sharpener/attackby(obj/item/I, mob/user, params)
+	if(used)
+		user << "<span class='notice'>The sharpening block is too dull to use again.</span>"
```
Comment (neersighted) [inline code/game/objects/items/weapons/sharpener.dm]: whetstone's don't go dull... how about 'too worn'?

File: code/game/objects/items/weapons/sharpener.dm [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+		return
+	if(I.force >= max || I.throwforce >= max)
+		user << "<span class='notice'>[I] is much too powerful to sharpen further.</span>"
+		return
+	if(istype(I, /obj/item/weapon))
+		user << "<span class='notice'>You sharpen [I] with [src], making it much more deadly than before.</span>"
+		I.sharpness = IS_SHARP
+		I.force = (I.force + increment)
```
Comment (neersighted) [inline code/game/objects/items/weapons/sharpener.dm]: `I.force += increment`

File: code/game/objects/items/weapons/sharpener.dm [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+		I.throwforce = (I.throwforce + increment)
```
Comment (neersighted) [inline code/game/objects/items/weapons/sharpener.dm]: ditto

File: code/game/objects/items/weapons/sharpener.dm (hunk 2 above)
Comment (neersighted) [inline code/game/objects/items/weapons/sharpener.dm]: Yeah replace everything here with clamp please

**Comment by KazeEspada** [comment]
:-1: makes every weapon a cleaver without nerfing them somewhere else and doesnt give a cleaver or fireaxe anything in return. COMPLETELY UTTERLY DEDTROYS WEAPON BALANCE.

**Comment by drynwyn** [comment]
You can get more then 2 by using vending machine restock modules

**Comment by Ultimate-Chimera** [comment]
I'm gonna lean for a :-1: on this.

**Comment by Incoming5643** [comment]
Sharpened Sharpened Sharpened Major space carp plushie of the world +15

**Comment by PKPenguin321** [comment]
balance police have arrived :rotating_light: :police_car: :cop: :oncoming_police_car: :rotating_light:

**Comment by Zilenan91** [comment]
Additionally, you can also sharpen those chaplain staves twice to get a 15 brute, 50% block for everything murder machine.

**Comment by PKPenguin321** [comment]
you can no longer sharpen items with block chance

**Comment by Lumipharon** [comment]
Sharpened sharpened bike horn.

Honk.

**Comment by Zilenan91** [comment]
@PKPenguin321 Vend both sharpening blocks and nothing else then deconstruct the machine. It should drop all of the components needed to remake it, and once created, be left with 1 sharpening block and 1 less of something else.

**Comment by Lumipharon** [comment]
@Zilenan91 I can't say for kitchen vendors, but many vendors (such as engie vendors) canot be deconstructed, and so cannot be abused in this way.

**Comment by AlexanderUlanH** [comment]
15 is the gold standard for practical (read: normal-size) weapons that're widely available so yeah this is pretty unbalanced. I'm all for goofy stuff, but I don't think this is worth having.

**Comment by PKPenguin321** [comment]
If 15 proves to be too much in the future (I genuinely do not think it will), we can always lower the max. It's as simple as lowering a var.

**Comment by Zilenan91** [comment]
It's fine

**Comment by Fox-McCloud** [comment]
Another concern: base 15 melee weapons with subtle hitsounds *and* no in-hand icon.

Things that deal loads of damage generally have fairly distinctive hitsounds and have obvious in-hand icons and/or deal fire damage (meaning no chance to knock down).

**Comment by KazeEspada** [comment]
No... 10 is the golden standard. 15 is above average.
Screwdriver=15 against people without eye protection.
15=Cleaver
15=Welder
For example:
Toolboxes=10
Fire Extringuishers =10
Most tools=7
Floor tiles=10

File: code/game/objects/items/weapons/sharpener.dm [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+	if(used)
+		user << "<span class='notice'>The sharpening block is too worn to use again.</span>"
+		return
+	if(I.force >= max || I.throwforce >= max)
+		user << "<span class='notice'>[I] is much too powerful to sharpen further.</span>"
+		return
+	if(I.block_chance > 0)
+		user << "<span class='notice'>Sharpening [I] would make it completely ineffective at blocking. You decide against it.</span>"//blame zilenan
+		return
+	if(istype(I, /obj/item/weapon))
```
Comment (KazeEspada) [inline code/game/objects/items/weapons/sharpener.dm]: Except ironically not all weapons are weapons.

**Comment by Lumipharon** [comment]
>sharpened sharpened paper

Death by papercut, even better then death by honk.

File: code/game/objects/items/weapons/sharpener.dm (hunk 11 above)
Comment (PKPenguin321) [inline code/game/objects/items/weapons/sharpener.dm]: as intended, my friend

**Comment by PKPenguin321** [comment]
>Things that deal loads of damage generally have fairly distinctive hitsounds and have obvious in-hand icons and/or deal fire damage (meaning no chance to knock down).

could always make the block give them a much louder hitsound (like the knife hitsound). inhands are weird, though.

**Comment by Zilenan91** [comment]
Or just force them all to have generic slicing sounds like the cleaver.

**Comment by PKPenguin321** [comment]
yeah, that's what i meant

**Comment by Zilenan91** [comment]
Those tricks work with eveyr other machine, I guess the kitchen vendors don't have circuit boards to recreate them. Try it on the bar vendor or a snack vendor or something.

**Comment by Jalleo** [comment]
Sharpened police baton?

**Comment by Razharas** [comment]
Maybe instead of doing silly stuff like sharpened fire extinguishers make it so only items with sharp flag can be sharpened further?
As of right now i dont like it

**Comment by Xhuis** [comment]
Cool. :+1:

**Comment by KorPhaeron** [comment]
@Razharas 

something like

if (I.sharpness = IS_SHARP)
I.sharpness = IS_SHARP_PRECISE

And you can't sharpen precise items any further? Would take care of the issue of stacking items, and would allow you to sharpen things that make sense (like fire-axes or swords).

Though the chef would get his force 20 cleaver roundstart so I dunno

**Comment by GunHog** [comment]
He stated that it only works on items with less than 15 force, so that is not a concern.

**Comment by RemieRichards** [comment]
@KorPhaeron equivalence is two equals operators, ```==```

**Comment by tkdrg** [comment]
yeah what razh said, this is dumb for the same reason the deep frier is dumb

**Comment by Zilenan91** [comment]
The deep frier was fine, the entire playerbase wanted it.

**Comment by KazeEspada** [comment]
deep frier doesn't get me killed by a force 10 pen.

**Comment by Zilenan91** [comment]
>you feel a massive prick

**Comment by KazeEspada** [comment]
sharpened plasma tank =15 force
sharpened kitchen knife  = 15 force and fits in your pocket
sharpened sharpened nuke disc = 10 force, no sprite
sharpened sharpened pen = 10 force, no sprite
sharpened sharpened screwdriver = 15 force, tiny sprite
sharpened sharpened fork = 15 force tiny sprite
sharpened sharpened paper = 10 force, no sprite
sharpened sharpened syndicate balloon = 10 force, ITS A FUCKING BALLOON
sharpened sharpened banana peel = 10 force, How?

**Comment by Zilenan91** [comment]
It's fine man. Realistically the chefs are only going to sharpen their knives with this and leave it be.

**Comment by Xhuis** [comment]
> sharpened sharpened fork: 15 force tiny sprite

throwback to forklander

**Comment by Steelpoint** [comment]
>sharpened pen

Let. It. Begin.

**Comment by PKPenguin321** [comment]
>sharpened sharpened syndicate balloon = 10 force, ITS A FUCKING BALLOON
sharpened sharpened banana peel = 10 force, How?

it's really sharp, duh

**Comment by Cheridan** [comment]
both poorly implemented and a bad idea

durf sharpened sharpened latex gloves
hurr tryhard shitcurity getting these and having sharpened sharpened stun batons

**Comment by PKPenguin321** [comment]
Can't sharpen clothing like gloves I'm pretty sure, they do not fall under /weapon.

Stun batons could only ever be sharpened once due to the 15 cap, and I'd hope that the crew would use the blocks before sec got to it (but of course this wont always go the way it should).

I'm not sure how I could implement them better short of putting them directly onto the map rather than into the vendor.

The idea of sharpened objects being funny or not funny is entirely subjective, and many people in this very PR have expressed excitement for sharpening the silliest things.

Please work with me instead of slamming me with hurrs and durrs. :(

**Comment by octareenroon91** [comment]
Consider checking for brute damage type.

Sharpening can turn a precise sharp thing into just sharp. I can see that being balanced, so I don't know if you should change it.

Consider using a blue/gray recolor of the sandstone block.

**Comment by Cheridan** [comment]
Sorry for being rude.

Still, the /weapon subtype includes _many many_ things that are not weapons. Like the aforementioned banana peel.
"sharpened sharpened item" is bad implementation. It should be something like "twice-sharpened item" or just "sharpened item". Applying the prefix twice just comes off as laziness.

Now, I want you to ask yourself what this, conceptually, adds to the game. What's the benefit of it? The things I see it doing are:
- Adds more absurdity. Benefit for some people, negative for others.
- Adds a powergaming element. The people who aren't sharpening sunflowers for laughs are going to be looking for the best thing to sharpen to give them an advantage.
- Invalidating existing items. Chef gets a 15 damage cleaver as a locked vendor item. Expect it to never be seen again since he can bring anything else up to that level (until the inevitable "BUFFS CLEAVER" pr).
-Buffs chef? Which, really, if that's the end goal just give him the cleaver roundstart and call it a day.

**Comment by PKPenguin321** [comment]
Thanks a lot for the genuine feedback!
>Adds more absurdity. Benefit for some people, negative for others.

I'm gonna gamble on the benefit for this one. As somebody that plays a lot, there are only maybe a handful of people that I can think of who would dislike this.

The other two points could probably be solved with lowering the max damage. Powergaming benefits would be minimized, and existing items wouldn't be quite as invalidated. I'm thinking 12 for the max damage, with increments of 4 per block. I'll whip up a new commit soon, I look forward to more constructive feedback like yours. :smiley:

**Comment by PKPenguin321** [comment]
@Cheridan I have addressed much of your feedback, and feel this is ready for a final review/merge.

**Comment by Ultimate-Chimera** [comment]
Oh god no.

**Comment by PKPenguin321** [comment]
>Oh god no.

Nice feedback, friend! It sure means a lot.

**Comment by phil235** [comment]
:-1: The idea of making sharp items extra sharp with a sharpener isn't necessarily a bad one, but this is just silly. I completely agree with Kaze and Cheri's comments.

**Comment by PKPenguin321** [comment]
@Cheridan I need the *final judgement*
Close or merge, up to you

**Comment by ChangelingRain** [comment]
It shouldn't be able to sharpen things that have less than 5 force or the NOBLUDGEON flag because that's silly.

**Comment by phil235** [comment]
You could use an is_sharp() check so this only sharpen sharp items.

**Comment by Cheridan** [comment]
@PKPenguin321 If you add a check for sharp objects then I'll merge this.

**Comment by PKPenguin321** [comment]
*sigh*
fine, i'll do it tomorrow

**Comment by Zilenan91** [comment]
If you want to restrict it to sharp items can you make the blocks do +5 brute each time too?

**Comment by PKPenguin321** [comment]
that's what i'm thinking

**Comment by PKPenguin321** [comment]
big update, check the latest commit/the original post

**Comment by Zilenan91** [comment]
What practical use is there in setting the max sharpness to 30? I can't think of a single weapon that can get there with a single sharpen.

**Comment by PKPenguin321** [comment]
mainly just so you can't sharpen eswords tbh

**Comment by Zilenan91** [comment]
Don't eswords do 30 brute rather than 31? I guess it would still mean that you can't sharpen it? And if you did it would just waste it right.

**Comment by PKPenguin321** [comment]
you can't sharpen things that are 30 or higher, you can't sharpen an esword at all
unless for whatever reason it keeps sharpness when it's just a handle (which would be a bug on it's own), but the then sharpen would be wasted because when you turned it on it would set it back to 30

**Comment by Fox-McCloud** [comment]
This means that chef effectively has a 23 force weapon, every single shift.

Axes/spears don't matter because of how wielding them works--it'll just reset on the next wield.

The big winner in all of this, though, will be the chaplain--who will be able to attain a whopping 26 force weapon if he can get his hands on both the blocks.

**Comment by Zilenan91** [comment]
You can only sharpen things once

**Comment by Fox-McCloud** [comment]
>You can only sharpen things once

You can only use the block once--you can sharpen things, more than once, so long as it's under 30.

**Comment by Fox-McCloud** [comment]
>You can only sharpen things once

Yeah, updated the math on it.

**Comment by Zilenan91** [comment]
All of this is totally fine.

**Comment by PKPenguin321** [comment]
yeah and in reality the blocks will most likely be long used up before a fireaxe/chaplain weapon can get to it (unless the chef is a sneaky antag, in which case i have no problem with this)

**Comment by Fox-McCloud** [comment]
Hmm...thoughts on making it a contraband item?

**Comment by Zilenan91** [comment]
It's fine

**Comment by Lumipharon** [comment]
Holy fuck the discussion in this pr is still going.

**Comment by KorPhaeron** [comment]
>Holy fuck the discussion in this pr is still going.

**Comment by PKPenguin321** [comment]
>KazeEspada	You cant even sharpen knives
KazeEspada	because knives are IS_SHARP_ACCURATE
KazeEspada	not IS_SHARP

thanks for pointing that out in this PR before it got merged instead of in IRC after it got merged :sob:

**Comment by GunHog** [comment]
# TESTING

**Comment by PKPenguin321** [comment]
\>when you tested every part of your PR on every commit but the last one
\>when you test the last one but quiiiite thoroughly enough

i tested it with a spawned in fireaxe and a spawned in block because i had just implemented two handed shit (which worked) then tested it with a glass shard from a window i broke with the axe

both of these objects have IS_SHARP and it never occured to me to try something with IS_SHARP_ACCURATE

i'm a failure :sob:

**Pull request merged**